packages/eip712/src/encoding/decodeProtobuf.ts

```diff
@@ -0,0 +1,153 @@
+import { Registry } from '@cosmjs/proto-signing'
+import { AuthInfo, SignDoc, TxBody } from 'cosmjs-types/cosmos/tx/v1beta1/tx'
+import { MsgVote } from 'cosmjs-types/cosmos/gov/v1beta1/tx'
+import { MsgDelegate } from 'cosmjs-types/cosmos/staking/v1beta1/tx'
+import { parseChainId } from './utils'
+import { MSG_TYPES, eip712MessageType, getFeePayerFromMsg } from './decodeAmino'
+import { createEIP712, generateFee, generateMessage } from '../messages/base'
+
+const Long = require('long')
+const { isLong } = require('long')
+
+export const PROTO_MSG_TYPES = {
+  MSG_SEND: '/cosmos.bank.v1beta1.MsgSend',
+  MSG_VOTE: '/cosmos.gov.v1beta1.MsgVote',
+  MSG_DELEGATE: '/cosmos.staking.v1beta1.MsgDelegate',
+}
+
+function protobufTypeUrlToAminoType(typeUrl: string) {
```

Comment:
are we only supporting these msgs?

Reply:
No, we will add the remaining messages in a follow-up PR.